src/orion/plotting/backend_plotly.py

```diff
@@ -0,0 +1,103 @@
+import pandas as pd
+import plotly.graph_objects as go
+
+
+def regret(experiment, order_by, verbose_hover, **kwargs):
+
+    def template_trials():
+        template = '<b>ID: %{customdata[0]}</b><br>' \
+            'value: %{y}<br>' \
+            'time: %{customdata[1]|%Y-%m-%d %H:%M:%S}<br>'
+
+        if verbose_hover:
+            template += 'parameters: %{customdata[2]}'
+
+        template += '<extra></extra>'
+
+        return template
+
+    def template_best():
+        return '<b>Best ID: %{customdata[0]}</b><br>' \
+            'value: %{customdata[1]}' \
+            '<extra></extra>'
+
+    ORDER_KEYS = ['suggested', 'reserved', 'completed']
+
+    if not experiment:
+        raise ValueError("Parameter 'experiment' is None")
+
+    if order_by not in ORDER_KEYS:
+        raise ValueError(f"Parameter 'order_by' is not one of {ORDER_KEYS}")
+
+    trials = list(filter(lambda trial: trial.status == 'completed', experiment.fetch_trials()))
+
+    data = [(trial.id, trial.status, trial.submit_time, trial.start_time,
+             trial.end_time, format_hyperparameters(trial.params), trial.objective.value) for trial in trials]
+
+    df = pd.DataFrame(data, columns=['id', 'status', ORDER_KEYS[0],
+                                     ORDER_KEYS[1], ORDER_KEYS[2], 'params', 'objective'])
+
+    df = df.sort_values(order_by)
+
+    df['best'] = df['objective'].cummin()
+    df['best_id'] = get_best_ids(df)
```

Comment:
It would be good to extract the data preparation process. Experiment.to_frame() could even be a method would be useful in general.

Reply:
I'm happy to extract the data preparation process. That's a great suggestion! Since the preparation is strictly related to the plot (i.e., some columns are irrelevant outside of the plot), I'm not going to export it outside.

`Experiment.to_frame()` would be useful indeed. However, wouldn't it be more polyvalent if we pass a dict of the experiment to build a dataframe in a general case?

Reply:
It depends, if the data frame should contain the trials then the dictionary of the experiment would be insufficient. Also `some_experiment.to_frame()` would follow standard conversion methods in tools such as xarray. From what I see in `build_frame()` I would say these information (columns) are useful in general.